src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,36 @@
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        {
+            var optionalIfElements = page.Elements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIfValue) 
+                || !string.IsNullOrEmpty(element.Properties.OptionalIfNotValue));
+
+            if (optionalIfElements.Any())
+            {
+                foreach (var element in optionalIfElements)
+                {
+                    var userChoice = convertedAnswers.AllAnswers
+                        .Where(answer => answer.QuestionId.Equals(element.Properties.OptionalIfQuestionId))
+                        .FirstOrDefault();
+
+                    if (userChoice is not null)
```

Comment:
Could invert this too: if null > continue through the loop. Just reduces that nesting...